crates/ruff_linter/src/rules/pylint/rules/too_few_public_methods.rs

```diff
@@ -0,0 +1,87 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast as ast;
+use ruff_python_semantic::analyze::visibility::{self, Visibility::Public};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for classes that only have a public `__init__` method,
+/// as well as 0 base classes, and 0 decorators.
+///
+/// ## Why is this bad?
+/// Classes with just an `__init__` are possibly better off
+/// being a dataclass or a namedtuple, which are more lightweight.
+///
+/// ## Example
+/// ```python
+/// class Point:
+///     def __init__(self, x: float, y: float):
+///         self.x = x
+///         self.y = y
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from dataclasses import dataclass
+///
+///
+/// @dataclass
+/// class Point:
+///     x: float
+///     y: float
+/// ```
+#[violation]
+pub struct TooFewPublicMethods;
+
+impl Violation for TooFewPublicMethods {
```

Comment:
Hmm, you're saying it's complicated ;)

I would prefer being more strict and setting the baseline to what's required for data classes (type annotations), considering that the fields used to be named. It otherwise becomes necessary to suppress the rule if users don't want to use a tuple, but can't use a data class.